lib/workers/repository/process/vulnerabilities.ts

```diff
@@ -0,0 +1,120 @@
+import { Ecosystem, Osv, OsvOffline } from '@jamiemagee/osv-offline';
+import pAll from 'p-all';
+import { getManagerConfig, mergeChildConfig } from '../../../config';
+import type { PackageRule, RenovateConfig } from '../../../config/types';
+import { logger } from '../../../logger';
+import type {
+  PackageDependency,
+  PackageFile,
+} from '../../../modules/manager/types';
+
+export class Vulnerabilities {
+  private osvOffline: OsvOffline;
+
+  private static readonly managerEcosystemMap: Record<string, Ecosystem> = {
+    npm: 'npm',
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  private constructor() {}
+
+  private async initialize(): Promise<void> {
+    this.osvOffline = await OsvOffline.create();
+  }
+
+  static async create(): Promise<Vulnerabilities> {
+    const instance = new Vulnerabilities();
+    await instance.initialize();
+    return instance;
+  }
+
+  async fetchVulnerabilities(
+    config: RenovateConfig,
+    packageFiles: Record<string, PackageFile[]>
+  ): Promise<void> {
+    const managers = Object.keys(packageFiles).filter(
+      (manager) => Vulnerabilities.managerEcosystemMap[manager] !== undefined
+    );
+    const allManagerJobs = managers.map((manager) =>
+      this.fetchManagerVulnerabilities(config, packageFiles, manager)
+    );
+    await Promise.all(allManagerJobs);
+  }
+
+  private async fetchManagerVulnerabilities(
+    config: RenovateConfig,
+    packageFiles: Record<string, PackageFile[]>,
+    manager: string
+  ): Promise<void> {
+    const managerConfig = getManagerConfig(config, manager);
+    const queue = packageFiles[manager].map(
+      (pFile) => (): Promise<void> =>
+        this.fetchManagerPackagerFileUpdates(config, managerConfig, pFile)
+    );
+    logger.trace(
+      { manager, queueLength: queue.length },
+      'fetchManagerUpdates starting'
+    );
+    await pAll(queue, { concurrency: 5 });
+    logger.trace({ manager }, 'fetchManagerUpdates finished');
+  }
+
+  private async fetchManagerPackagerFileUpdates(
+    config: RenovateConfig,
+    managerConfig: RenovateConfig,
+    pFile: PackageFile
+  ): Promise<void> {
+    const { packageFile } = pFile;
+    const packageFileConfig = mergeChildConfig(managerConfig, pFile);
+    const { manager } = packageFileConfig;
+    const queue = pFile.deps.map(
+      (dep) => (): Promise<PackageRule[]> =>
+        this.fetchDependencyVulnerabilities(packageFileConfig, dep)
+    );
+    logger.trace(
+      { manager, packageFile, queueLength: queue.length },
+      'fetchManagerPackagerFileUpdates starting with concurrency'
+    );
+
+    config.packageRules.push(...(await pAll(queue, { concurrency: 5 })).flat());
+    logger.trace({ packageFile }, 'fetchManagerPackagerFileUpdates finished');
+  }
+
+  private async fetchDependencyVulnerabilities(
+    packageFileConfig: RenovateConfig & PackageFile,
+    indep: PackageDependency
```

Comment:
Would you consider a clearer/more speaking names?

Reply:
I copied most of this from https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/process/fetch.ts

But refactoring now, before anything is calling this code, is definitely an option. Do you mean `indep` specifically? Or are there others?

Reply:
Just `indep`. I don't understand what it means.